modules/datastore/modules/datastore_mysql_import/src/Service/MysqlImport.php

```diff
@@ -86,38 +97,32 @@ protected function runIt() {
       return $this->setResultError(sprintf('Unable to resolve file name "%s" for resource with identifier "%s" and version "%s".', $file_path, $this->resource->getIdentifier(), $this->resource->getVersion()));
     }
 
-    // Read the header line from the CSV file.
+    // Read the columns and EOL character sequence from the CSV file.
     try {
-      $header_line = $this->getFirstLineFromFile($file_path);
+      ['columns' => $columns, 'lines' => $column_lines] = $this->getColsFromFile($file_path);
     }
     catch (FileException $e) {
       return $this->setResultError($e->getMessage());
     }
-
-    // Extract the columns names using the header line.
-    $columns = str_getcsv($header_line);
+    // Attempt to detect the EOL character sequence for this file; default to
+    // '\n' on failure.
+    $eol = $this->getEol($column_lines) ?? '\n';
+    // Count the number of EOL characters in the header row to determine how
+    // many lines the headers are occupying.
+    $header_line_count = substr_count(trim($column_lines), self::EOL_TABLE[$eol]) + 1;
     // Generate sanitized table headers from column names.
     $headers = $this->generateTableHeaders($columns);
-
     // Use headers to set the storage schema.
     $this->setStorageSchema($headers);
 
-    // Attempt to detect the line ending for this resource file using the first
-    // line from the file.
-    $eol = $this->getEol($header_line);
-    // On failure, stop the import job and log an error.
-    if (!isset($eol)) {
-      return $this->setResultError(sprintf('Failed to detect EOL character for resource file "%s" from header line "%s".', $file_path, $header_line));
-    }
-
     // Call `count` on database table in order to ensure a database table has
     // been created for the datastore.
     // @todo Find a better way to ensure creation of datastore tables.
     $this->dataStorage->count();
     // Construct and execute a SQL import statement using the information
     // gathered from the CSV file being imported.
     $this->getDatabaseConnectionCapableOfDataLoad()->query(
-      $this->getSqlStatement($file_path, $this->dataStorage->getTableName(), $headers, $eol));
+      $this->getSqlStatement($file_path, $this->dataStorage->getTableName(), $headers, $eol, $header_line_count));
```

Comment:
```suggestion
      $this->getSqlStatement($file_path, $headers, $eol, $header_line_count));
```
CodeClimate's complaining about too many arguments, and this can be done inside the method.